File: backend/models/summarization.py
from pydantic import UUID4, BaseModel, ConfigDict, Field
from typing import Optional
from datetime import datetime

from models.uploaded_file import UploadedFileDTO


class SummarizationBase(BaseModel):
    id: UUID4
    file_id: UUID4
    summary_text: str
    ai_model_used: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(str_max_length=255)


class SummarizationDTO(SummarizationBase):
    file: UploadedFileDTO

File: frontend/src/types/models/summarization.ts
import type { UploadedFile } from "./uploaded-file";
import type { Recommendation } from "./recommendation";

export type Summarization = {
  id: string;
  file_id: string;
  name: string;
  summary_text: string;
  ai_model_used: string | undefined;
  created_at: string;
  updated_at: string;
  file: UploadedFile;
  recommendations: Recommendation[];
};
